proxywasm/hostcall.go

```diff
@@ -101,27 +107,33 @@ func GetDownStreamData(start, maxSize int) ([]byte, error) {
 	return ret, types.StatusToError(st)
 }
 
-func SetDownStreamData(body []byte) error {
-	var bufferData *byte
-	if len(body) != 0 {
-		bufferData = &body[0]
-	}
-	st := rawhostcall.ProxySetBufferBytes(types.BufferTypeDownstreamData, 0, len(body), bufferData, len(body))
-	return types.StatusToError(st)
+func AppendDownStreamData(data []byte) error {
+	return appendToBuffer(types.BufferTypeDownstreamData, data)
+}
+
+func PrependDownStreamData(data []byte) error {
+	return prependToBuffer(types.BufferTypeDownstreamData, data)
+}
+
+func ReplaceDownStreamData(data []byte) error {
+	return replaceBuffer(types.BufferTypeDownstreamData, data)
 }
 
 func GetUpstreamData(start, maxSize int) ([]byte, error) {
 	ret, st := getBuffer(types.BufferTypeUpstreamData, start, maxSize)
 	return ret, types.StatusToError(st)
 }
 
-func SetUpstreamData(body []byte) error {
-	var bufferData *byte
-	if len(body) != 0 {
-		bufferData = &body[0]
-	}
-	st := rawhostcall.ProxySetBufferBytes(types.BufferTypeUpstreamData, 0, len(body), bufferData, len(body))
-	return types.StatusToError(st)
+func AppendUpstreammData(data []byte) error {
```

Comment:
typo: `AppendUpstreammData` -> `AppendUpstreamData`

Reply:
thanks! resolved in https://github.com/tetratelabs/proxy-wasm-go-sdk/pull/175